code/presentation/mb-client/src/screens/home/masjid-screen.tsx

```diff
@@ -0,0 +1,123 @@
+/* eslint-disable max-lines-per-function */
+import { useNavigation, useRoute } from '@react-navigation/native';
+import React, { useState } from 'react';
+import { Button, StyleSheet, Text, View } from 'react-native';
+
+import { useMasjids } from '@/api/masjid/use-masjids';
+import { getNextNamaz } from '@/api/masjid/util';
+import { masjid } from '@/core/supabase';
+import type { RouteProp } from '@/navigation/types';
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+    backgroundColor: '#333',
+  },
+  text: {
+    color: 'green',
+    marginBottom: 10,
+    fontSize: 18,
+    fontWeight: 'bold',
+    borderWidth: 1,
+    borderColor: 'green',
+    padding: 10,
+    borderRadius: 5,
+  },
+});
+
+const MasjidScreen = () => {
+  const { data: masjidsWithDistance, isLoading, isError } = useMasjids();
+  const { params } = useRoute<RouteProp<'MasjidScreen'>>();
+  const selectedMasjidId = params.selectedMasjidId;
+  const navigation = useNavigation();
```

Comment:
as we are just using navigate function, we can do ` const { navigate } = useNavigation();` to be consistent with the existing code base